ivy/functional/backends/numpy/activations.py

```diff
@@ -16,38 +17,54 @@ def relu(x: np.ndarray, out: Optional[np.ndarray] = None) -> np.ndarray:
     return np.maximum(x, 0, out=out)
 
 
-def leaky_relu(x: np.ndarray, alpha: Optional[float] = 0.2) -> np.ndarray:
-    return np.where(x > 0, x, x * alpha)
+def leaky_relu(
+    x: np.ndarray,
+    alpha: Optional[float] = 0.2,
+    out: Optional[np.ndarray] = None
+) -> np.ndarray:
+    ret = np.asarray(np.where(x > 0, x, x * alpha), x.dtype)
+    if ivy.exists(out):
```

Comment:
we should not handle `out` in the backend. We should remove the `out` argument here unless the backend function itself is able to handle this for us.